pylint/plugins/hass_enforce_type_hints.py

```diff
@@ -18,10 +18,9 @@
 class TypeHintMatch:
     """Class for pattern matching."""
 
-    module_filter: re.Pattern
     function_name: str
     arg_types: dict[int, str]
-    return_type: list[str] | str | None
+    return_type: list[str] | str | None | object
 
 
 _TYPE_HINT_MATCHERS: dict[str, re.Pattern] = {
```

Comment:
```suggestion
_TYPE_HINT_MATCHERS: dict[str, re.Pattern[str]] = {
```
Not directly related to this PR, but `re.Pattern` is generic and could be typed as `re.Pattern[str]`. Similarly below for `_MODULE_FILTERS`.

Reply:
I'll adjust that in a follow-up PR, together with the astroid import

Reply:
I changed my mind and implemented this now (not in follow-up PR)